paasta_tools/setup_kubernetes_job.py

```diff
@@ -146,13 +147,17 @@ def setup_kube_deployments(
             elif app.kube_deployment not in existing_kube_deployments:
                 log.debug(f"Updating {app} because configs have changed.")
                 app.update(kube_client)
-            elif autoscaling_is_paused() and not is_deployment_marked_paused(
-                kube_client, app.soa_config
+            elif (
```

Comment:
I'm a little worried that this approach is going to be too brittle.  What if we add another call to `is_deployment_marked_paused` somewhere else and forget the `isinstance` gate?  I guess my preference would be to add a `try...catch` block to `is_deployment_marked_paused` itself so that if we try to call it on something that's not a deployment we can fail gracefully.  This isn't a strong preference, but just my two cents.

As an aside, another reason I suggest this is because I think this block of if/else's is already pretty cumbersome, and I'm wary of making it more-so.

Reply:
I ended up renaming `get_annotations_for_deployment` to `get_annotations_for_kubernetes_service`, and takes both stateful sets and deployments into consideration in its implementation.